types/cookie-parser/index.d.ts

```diff
@@ -7,6 +7,20 @@
 
 import * as express from 'express';
 
+declare global {
+    namespace Express {
+        // Inject additional properties on express.Request
+        interface Request {
+            /**
+             * This request's secret.
+             * Optionally set by cookie-parser if secret(s) are provided.  Can be used by other middleware.
+             * [Declaration merging](https://www.typescriptlang.org/docs/handbook/declaration-merging.html) can be used to add your own properties.
+             */
+            secret?: string;
```

Comment:
There are more properties added to the Request object:

- `cookies: Record<string, string>`
- `signedCookies: Record<string, string | Record<string,unknown> | false>`

Reply:
They are technically part of the core express request object here, but as "any" type.  I wasn't sure if desire would be to override the types here, but would happy to add and try it out.

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express-serve-static-core/index.d.ts#L622
https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/express-serve-static-core/index.d.ts#L632

Thoughts?

Reply:
Yeah, go for it. Ping me if you encounter any problems.

Reply:
Thanks for help, I just pushed the defs for cookies/signedCookies, but test is failing locally.  Two issues:

1. It seems to inherit the jsdoc, but not the more specific types.  Still returning as `any` in the test (and VS Code).
1. Are we sure the types are specific enough, given that JSON cookies are a possibility as well?  e.g. `Record<string, string>` seems like would be too restrictive for `cookies`.

Let me know if you have any thoughts on how to override the types (and ensure they are correct), or if something should that be discussed in `express-service-static-core`?

Reply:
You're absolutely right, `cookies` must have the type `Record<string, unknown>`. Basically, `signedCookies` should have the same type. But this will break lots of users. Now that I think about it, just remove the two, this is too big and too breaking.